.teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt

```diff
@@ -91,7 +91,7 @@ class CIConfigIntegrationTests {
                 }
 
                 // hacky way to consider deferred tests
-                val deferredTestCount = if (stage.stageName == StageNames.READY_FOR_NIGHTLY) 4 else 0
+                val deferredTestCount = 0
```

Comment:
I was thinking about that too. Are you sure you/we don't want to use this again in the future for something else. Looks like some work went into supporting this (although I don't know about the quality of that implementation).

If you think this can go, I can do it, but I would do it in a separate step after merging this PR.

Reply:
Yeah, let's do it as a separate step. We can also leave it in for now. I really don't like using a static mutable field in the implementation to capture the deferred tests. We can leave it also in for now and make it better when we start using it again.